x-pack/legacy/plugins/siem/public/components/ml/tables/anomalies_host_table.tsx

```diff
@@ -51,35 +54,39 @@ export const AnomaliesHostTable = React.memo<AnomaliesTableProps>(
       pageSizeOptions: [5, 10, 20, 50],
       hidePerPageOptions: false,
     };
-    return (
-      <EuiPanel>
-        <BasicTableContainer>
-          {loading && (
-            <>
-              <BackgroundRefetch />
-              <LoadingPanel
-                height="100%"
-                width="100%"
-                text={`${i18n.LOADING} ${i18n.ANOMALIES}`}
-                position="absolute"
-                zIndex={3}
-                data-test-subj="anomalies-host-table-loading-panel"
-              />
-            </>
-          )}
-          <HeaderPanel
-            subtitle={`${i18n.SHOWING}: ${hosts.length.toLocaleString()} ${i18n.ANOMALIES}`}
-            title={i18n.ANOMALIES}
-          />
-          <EuiInMemoryTable
-            items={hosts}
-            columns={columns}
-            pagination={pagination}
-            sorting={sorting}
-          />
-        </BasicTableContainer>
-      </EuiPanel>
-    );
+    if (!hasMlUserPermissions(capabilities)) {
+      return <></>;
```

Comment:
you can just return null

Reply:
Updated each part of the code with null and the TypeScript return types to include the `null` now.